src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int firstLeg;
+    private int secondLeg;
+    private int top;
+    private int bottom;
+
+    public IsoscelesTrapezoid(int firstLeg, int secondLeg, int top, int bottom, String color) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+        this.top = top;
+        this.bottom = bottom;
+    }
+
+    public int getFirstLeg() {
+        return firstLeg;
+    }
+
+    public int getSecondLeg() {
+        return secondLeg;
+    }
+
+    public int getTop() {
+        return top;
+    }
+
+    public int getBottom() {
+        return bottom;
+    }
+
+    @Override
+    public double getArea() {
+        if (getFirstLeg() != getSecondLeg() && getTop() < getBottom()) {
+            System.out.println("This is not isosceles ");
+            return 0;
+        }
```

Comment:
The condition for checking if the trapezoid is isosceles is incorrect. An isosceles trapezoid should have equal non-parallel sides (legs), but the current check only prints a message and returns 0 if the sides are not equal. Consider revising this logic to ensure it correctly identifies an isosceles trapezoid.